File: Domain/Entities/BoothChecklist.cs
namespace Nirvachak_AI.Domain.Entities;

/// <summary>
/// Records election-day readiness checks per booth (one per booth per election day).
/// </summary>
public class BoothChecklist
{
    public int Id { get; set; }
    public int BoothNumber { get; set; }
    public int ConstituencyId { get; set; }
    public Constituency? Constituency { get; set; }

    // Readiness items
    public bool AgentPresent        { get; set; }
    public bool BannerDisplayed     { get; set; }
    public bool VoterListPrinted    { get; set; }
    public bool TransportArranged   { get; set; }
    public bool PhoneCharged        { get; set; }
    public bool BoothClean          { get; set; }

    public string? Notes            { get; set; }
    public string? SubmittedByUserId { get; set; }
    public string? SubmittedByName   { get; set; }
    public DateTime SubmittedAt      { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt       { get; set; }
}

File: mobile/src/api/boothChecklist.ts
import apiClient from './client';

export interface ChecklistItem {
  boothNumber: number;
  boothName: string;
  address?: string | null;
  wardNumber?: string | null;
  assignedAgentName?: string | null;
  assignedAgentPhone?: string | null;
  agentPresent: boolean;
  bannerDisplayed: boolean;
  voterListPrinted: boolean;
  transportArranged: boolean;
  phoneCharged: boolean;
  boothClean: boolean;
  notes?: string | null;
  submittedByName?: string | null;
  submittedAt?: string | null;
  updatedAt?: string | null;
  isReady: boolean;
}

export interface ChecklistResponse {
  total: number;
  ready: number;
  items: ChecklistItem[];
}

export interface SaveChecklistRequest {
  boothNumber: number;
  agentPresent: boolean;
  bannerDisplayed: boolean;
  voterListPrinted: boolean;
  transportArranged: boolean;
  phoneCharged: boolean;
  boothClean: boolean;
  notes?: string | null;
}

export const getBoothChecklist = async (): Promise<ChecklistResponse> => {
  const { data } = await apiClient.get<ChecklistResponse>('/boothchecklist');
  return data;
};

export const saveBoothChecklist = async (req: SaveChecklistRequest): Promise<void> => {
  await apiClient.post('/boothchecklist', req);
};
